web.Dockerfile

```diff
@@ -0,0 +1,12 @@
+ARG IMAGE_NAME
+FROM ${IMAGE_NAME}
+# Based on https://code.visualstudio.com/remote/advancedcontainers/add-nonroot-user#_change-the-uidgid-of-an-existing-container-user
+# Without this, it will chown everything to www-data on the host and break permissions.
+RUN groupmod --gid 1000 www-data \
+    && usermod --uid 1000 --gid 1000 www-data \
+    && chown -R 1000:1000 /opt/web2py/applications
```

Comment:
Updated the comment to be more precise. It wouldn't be a full solution to run web2py as `www-data`, because it would still make the repo clone uneditable by the host user (that's what I meant by "break permissions"). It would also work if the upstream container used a new user with UID 1000.